### PR TITLE
Never load NFTs if no datasource

### PR DESCRIPTION
**Pull request by tomquirk**
This is a bug atm: 

scenario:
- user has a v2 and v3 FC
- V3 FC has NFTs
- User is on the `reconfigure FC` page in settings
- User switches from V3 to V2

Bug: V3 NFTs are still loaded and shown in Review and Deploy.

The bug is that, for some unknown reason, `tiersOf` is still returning the V3 tiers even when the datasource being sent in args is `0x00`.

### fix

This PR adds a failsafe to ensure the NFT context will never be populated without a valid (non-zero) datasource. Not great but works.

### Also

- removes some dead unused code related to NFTs (`yarn find-deadcode | grep nft`)

### CODE REVIEW

File: src/hooks/NftRewards.ts [hunk 1]
```diff
@@ -50,11 +50,12 @@ async function getRewardTierFromIPFS({
 export default function useNftRewards(
   tiers: JB721TierParams[],
   projectId: number | undefined,
+  dataSourceAddress: string | undefined,
 ): UseQueryResult<NftRewardTier[]> {
   const hasTiers = Boolean(tiers?.length)
 
   return useQuery(
-    ['nft-rewards', projectId],
```
Comment (tomquirk) [inline src/hooks/NftRewards.ts:57]: this is nice to have but doesn't actually fix this specific bug. but we should still have it here.

File: src/hooks/Transactor.ts [hunk 2]
```diff
@@ -35,8 +35,13 @@ function logTx({
       {},
     )
 
+  const log = [
+    `functionName=${functionName}`,
+    `contractAddress=${contract.address}`,
```
Comment (tomquirk) [inline src/hooks/Transactor.ts:40]: add contractAddress to log for more context

File: src/providers/v2v3/NftRewardsProvider.tsx [hunk 3]
```diff
@@ -6,31 +6,37 @@ import { useNftRewardTiersOf } from 'hooks/JB721Delegate/contractReader/NftRewar
...
    * Load NFT Rewards data
    */
   const { data: nftRewardTiersResponse, loading: nftRewardsCIDsLoading } =
     useNftRewardTiersOf({
-      dataSourceAddress: dataSourceAddress,
+      dataSourceAddress,
+      shouldFetch,
     })
+  // catchall to ensure nfts are never loaded if hasNftRewards is false (there's no datasource).
+  const tierData = shouldFetch ? nftRewardTiersResponse ?? [] : []
```
Comment (tomquirk) [inline src/providers/v2v3/NftRewardsProvider.tsx:28]: this is the actual fix. even if dataSource is 0 or undefined, there's some sketchy race condition or cache issue where `nftRewardTiersResponse` is still defined sometimes.

This line will set it to [] if there's no datasource (failsafe, hopefully).

File: src/utils/nftRewards.ts [hunk 4]
```diff
@@ -73,19 +73,6 @@ export function getNftRewardTier({
   return nftReward
 }
 
-// Sorts array of nft reward tiers by contributionFloor
```
Comment (tomquirk) [inline src/utils/nftRewards.ts:76]: unused these days.

File: src/providers/v2v3/NftRewardsProvider.tsx (hunk 3 above)
Comment (johnnyd-eth) [reply src/providers/v2v3/NftRewardsProvider.tsx:28]: Nice ok, good failsafe

File: src/utils/nftRewards.ts (hunk 4 above)
Comment (johnnyd-eth) [reply src/utils/nftRewards.ts:76]: Yep cool

**Review by johnnyd-eth — APPROVED** [review]
Nice one here @tomquirk. All LGTM

**Pull request merged**